staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/envelope.go

```diff
@@ -142,7 +203,7 @@ func (t *envelopeTransformer) TransformToStorage(data []byte, context value.Cont
 	// Append the length of the encrypted DEK as the first 2 bytes.
 	b := cryptobyte.NewBuilder(nil)
 	b.AddUint16LengthPrefixed(func(b *cryptobyte.Builder) {
-		b.AddBytes([]byte(encKey))
+		b.AddBytes(t.currentDEK.ciphertext)
```

Comment:
same as above - this should be protected by a read lock

Reply:
@awly PTAL - I think that the approach of locking the entire currentDEK may be more readable.